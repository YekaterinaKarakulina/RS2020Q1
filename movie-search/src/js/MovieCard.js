export default class MovieCard {
  constructor({
    title, year, poster, imdbID, linkToVideoGallery, imdbRating,
  }) {
    this.title = title;
    this.year = year;
    this.poster = poster;
    this.imdbID = imdbID;
    this.linkToVideoGallery = linkToVideoGallery;
    this.imdbRating = imdbRating;
  }

  createCardElement() {
    const cardElement = document.createElement('div');
    cardElement.className = 'swiper-slide';

    const cardHeader = document.createElement('a');
    cardHeader.className = 'card__header';
    cardHeader.setAttribute('href', this.linkToVideoGallery);
    cardHeader.text = this.title;

    const cardBody = document.createElement('div');
    cardBody.className = 'card__body';
    if (this.poster !== 'N/A') {
      cardBody.setAttribute('style', `background-image: url(${this.poster})`);
    } else {
      cardBody.innerText = 'Image not found';
    }

    const year = document.createElement('span');
    year.className = 'movieYear';
    year.innerHTML = `${this.year} year`;

    const rating = document.createElement('span');
    rating.className = 'movieRating';
    rating.innerHTML = `IMDb rating: ${this.imdbRating}`;

    const cardFooter = document.createElement('div');
    cardFooter.className = 'card__footer';
    cardFooter.append(year);
    cardFooter.append(rating);

    cardElement.append(cardHeader);
    cardElement.append(cardBody);
    cardElement.append(cardFooter);

    return cardElement;
  }
}
