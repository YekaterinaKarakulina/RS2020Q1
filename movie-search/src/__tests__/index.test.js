import MovieCard from '../js/MovieCard';

describe('MovieCard.createCardElement', () => {
  it('Should be instance of function', () => {
    expect(MovieCard.prototype.createCardElement).toBeInstanceOf(Function);
  });

  it('Should be render correctly', () => {
    const context = {
      title: 'Film',
      year: 2017,
      poster: 'linkToPoster',
      imdbID: 'id001',
      linkToVideoGallery: 'linkToVideogallery',
      imdbRating: 4.2,
    };
    document.body.appendChild(MovieCard.prototype.createCardElement.call(context));
    expect(document.body.innerHTML).toMatchSnapshot();
  });
});
