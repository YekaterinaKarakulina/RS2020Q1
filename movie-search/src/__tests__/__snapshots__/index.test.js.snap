// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard.createCardElement Should be render correctly 1`] = `"<div class=\\"swiper-slide\\"><a class=\\"card__header\\" href=\\"linkToVideogallery\\">Film</a><div class=\\"card__body\\" style=\\"background-image: url(linkToPoster)\\"></div><div class=\\"card__footer\\"><span class=\\"movieYear\\">2017 year</span><span class=\\"movieRating\\">IMDb rating: 4.2</span></div></div>"`;
